e2e/main_test.go

```diff
@@ -2,22 +2,57 @@ package e2e
 
 import (
 	"bytes"
+	"encoding/json"
 	"flag"
+	"io/ioutil"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"testing"
+
+	dockerConfigFile "github.com/docker/cli/cli/config/configfile"
 )
 
 var (
 	e2ePath         = flag.String("e2e-path", ".", "Set path to the e2e directory")
-	dockerApp       = os.Getenv("DOCKERAPP_BINARY")
-	dockerCli       = os.Getenv("DOCKERCLI_BINARY")
+	dockerCliPath   = os.Getenv("DOCKERCLI_BINARY")
 	hasExperimental = false
 	renderers       = ""
+	dockerCli       dockerCliCommand
 )
 
+type dockerCliCommand struct {
+	path         string
+	cliPluginDir string
+}
+
+func (d dockerCliCommand) createTestConfig() string {
+	configDir, err := ioutil.TempDir("", "config")
+	if err != nil {
+		panic(err)
+	}
+	config := dockerConfigFile.ConfigFile{CLIPluginsExtraDirs: []string{d.cliPluginDir}}
+	configFile, err := os.Create(filepath.Join(configDir, "config.json"))
+	if err != nil {
+		panic(err)
+	}
+	err = json.NewEncoder(configFile).Encode(config)
+	if err != nil {
+		panic(err)
+	}
+	err = os.Setenv("DOCKER_CONFIG", configDir)
+	if err != nil {
```

Comment:
We don't need to change the current process environment car but only impact the child process environment (setting this on the process level makes actually tests having huge side effects, preventing them to run in parallel in the future)

Reply:
The common rule for this kind of member function of a globally shared object is that it should not have side effects on the current process shared state (e.g. should not change process-wide environement variable).

It should return 2 values: 
- one string for the config dir
- one map[string]string usable to set child processes environment

Reply:
Instead of this I've renamed the function `createTestCmd`, it returns an `icmd.Command` with the `Env` prefilled and a cleanup function.